sensor_array/analysis/sample_analysis.py

```diff
@@ -457,34 +447,21 @@ def execute_sample_analysis(config_file):
         os.mkdir(results_filepath+folder)
 
         # ----- Loop over all breath samples -----
-        for i in range(len(all_breath_samples_joined)):
+        for i in range(len(all_breath_samples)):
 
             print('Breath Sample = ', i)
 
             # Load single breath sample
-            breath_sample = all_breath_samples_joined[i]
-            run_id = breath_sample['Run ID New']
-
-            # Get true comp and predicted mass
-            gases_for_true_comp = gases+['N2', 'O2']
-            true_comp = get_true_composoition(breath_sample, gases_for_true_comp)
-            true_comp_values = [true_comp[gas+'_comp'] for gas in gases]
+            breath_sample = all_breath_samples.loc[i:i]
```

Comment:
do you need .loc here? why i:i? 
did you know you can do a "join" in pandas? https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.join.html

Reply:
I do need loc for the same reason as before. Additionally, if I do df.loc[i] rather than df.loc[i:i] it returns a pandas series rather than a dataframe, and they behave slightly differently. There may be a reason to deal with these rather than dataframes, but I think I had an issues concatenating dataframes with series or vice-vera, so used this workaround.

Reply:
I think you don't need loc here for the same reason as above. Your general philosophy should be to "never use workarounds! (unless you have exhausted the search for the right way of doing it. You are almost never going to be the first person to need something)." In this case, you can do this in pandas: `all_breath_samples.loc[[i]]` which is the syntax for getting a dataframe. 

In general, you should use the series unless you NEED a dataframe. Not sure why you'd need a dataframe here...? Concatenating data frames should be exceedingly rare (maybe only when loading a dataset?). You should also be able to use an iterator here, such as DataFrame.iterrows. See https://stackoverflow.com/questions/16476924/how-to-iterate-over-rows-in-a-dataframe-in-pandas, both the answers that describe iterrows and the answer that says not to do it! (in your case, I think you may have to do it, but I haven't fully thought it through)